WIP: chore(repo): Update lint tools to what lektor uses

do3cc opened the pull request:
Still work in progress. Commits will be rebased!

Commit:
chore(repo): Update lint tools to what lektor uses

Still work in progress. Commits will be rebased!

do3cc commented:
No functional code changes. Apart from black changing everything left and right, I changed minor things in test code to make pylint happy.
I noticed Contributing docs is not complete, and I believe that setup.cfg and pyproject.toml could be unified, so I will try this later.
I only opened the PR as a draft to make anybody aware that a PR will come.

Commit:
WIP pyprojectification

dairiki commented:
There is some motivation to replace the use of black, flake8, reorder-python-imports, and pyupgrade with ruff (lektor/lektor#1209).  I'm not exactly sure how soon that may happen.  (I am swamped and not working on Lektor, atm.)

Anyway, there's really no reason that lektor-atom need use exactly the same tools as lektor.  And the use of ruff in place of the others is mostly compatible, anyway.

If you're going to go through this work now, it might be worth switching directly to ruff.

do3cc committed:
feat(repo): Use ruff

I cleaned up the configs and got rid of setup.cfg altogether.
To keep it simple, I use some very new features which will only be
available with the NEXT pip release in april.
I still consider this WIP. I want to test in a docker container and
extend CONTRIBUTING.md to explain both a pip and a uv workflow.

do3cc commented:
Thank you, that information was the reason I created the draft PR. I've updated to ruff, but in the next days I want to be sure everything works with both pip and uv.
